### ISSUE TITLE
After upgrade to latest version, lsp-javascript-typescript cannot find "lsp-make-traverser" from lsp-mode

### ISSUE THREAD

**Issue by cmatzenbach** [emacs-lsp/lsp-mode#507]
**Describe the bug**
I just upgraded my emacs packages, and now, every time I start emacs, I receive the following error:
`Error (use-package): lsp-javascript-typescript/:catch: Symbol’s function definition is void: lsp-make-traverser`
It looks like it knows something is wrong when trying to compile, I get the following message in the `*Compile-Log*`:
```
Leaving directory ‘/home/matzy/.emacs.d/elpa/lsp-javascript-typescript-20180614.2011’
Compiling file /home/matzy/.emacs.d/elpa/lsp-javascript-typescript-20180614.2011/lsp-javascript-typescript.el at Wed Dec  5 14:03:53 2018
Entering directory ‘/home/matzy/.emacs.d/elpa/lsp-javascript-typescript-20180614.2011/’
Compiling no file at Wed Dec  5 14:03:53 2018
Warning (bytecomp): the following functions are not known to be defined:
    lsp-make-traverser, lsp-provide-marked-string-renderer
```

**To Reproduce**
Upgrade to the latest version of lsp-mode and lsp-javascript-typescript.

**Expected behavior**
The package to load as normal and not throw errors upon startup.  I've been using this package without issue for close to a year now.

**Which Language Server did you use**
lsp-javascript-typescript

**OS**
Arch Linux

**Error callstack**
`Error (use-package): lsp-javascript-typescript/:catch: Symbol’s function definition is void: lsp-make-traverser`

Were there breaking API changes that came out in the past couple months?  My config has always worked without issue.  I also did `npm i -g javascript-typescript-langserver` to upgrade the actual server, but this seems outside the scope.  I do see the function in question (`lsp-make-traverser`) defined in lsp-common.el, but I cannot see it in M-x.  I set debug-on-error, and as I stepped through, I saw that the error is being thrown when lsp-javascript-typescript.el tries to call `lsp-make-traverser` on line 54.

**Comment by yyoncho** [comment]
We are about to move to new interface available in lsp.el - https://github.com/emacs-lsp/lsp-mode/blob/master/README-NEXT.md , currently we try to support both legacy lsp-mode.el and lsp.el but it looks like there is a bug. 

I believe the best option is to migrate to lsp.el . In order to do that, delete all the lsp-mode.el related configuration and replace it with:

```emacs-lisp
(require 'lsp-clinets)
(add-hook 'your-programming-lagnuage 'lsp)
```

`lsp` will automatically detect the presence of lsp-ui/company-lsp and it will configure it.

**Comment by cmatzenbach** [comment]
Ok, I tried this snippet:

```(use-package lsp-mode
  :defer t
  :config
  (require 'lsp-clients)
  (add-hook 'typescript-mode-hook 'lsp))```
When I open a file, it recognizes it's in a project and makes me select the root, but then I get the message: `"File mode specification error: (Couldn’t find executable typescript-language-server)"`.  What did I do wrong?

**Comment by yyoncho** [comment]
npm i -g javascript-typescript-langserver

**Comment by cmatzenbach** [comment]
@yyoncho  I already have that installed.  I've been using it for the past year without issues.

**Comment by seagle0128** [comment]
@yyoncho I think the langserver is incorrect in `lsp-clients.el`. 
I created a PR https://github.com/emacs-lsp/lsp-mode/pull/509 to address. Please review.

**Comment by yyoncho** [comment]
Fixed with #509
